stack/building-with-pip.rst

```diff
@@ -0,0 +1,241 @@
+Most science pipelines packages use the GPLv3 license and for those packages the ``pyproject.toml`` should use:
+
+.. code-block:: toml
+
+    license = {text = "GPLv3+ License"}
+    classifiers = [
+      "License :: OSI Approved :: GNU General Public License v3 or later (GPLv3+)",
+    ]
+
+Every Rubin DM science pipelines package should be owned by the special DM account and that should always be included in the ``authors`` section.
+Additional authors can be included if required.
+
+The ``dependencies`` section can only refer to packages that are available on `PyPI`_ since this is the section that will be read during ``pip install``.
+
+Setuptools Configuration
+------------------------
+
+For ``setuptools`` builds additional configuration is needed so that the python files and data files can be located.
+For example, in ``daf_butler`` there is this configuration:
+
+.. code-block:: toml
+
+    [tool.setuptools.packages.find]
+    where = ["python"]
+
+    [tool.setuptools]
+    zip-safe = true
+    license-files = ["COPYRIGHT", "LICENSE", "bsd_license.txt", "gpl-v3.0.txt"]
+
+    [tool.setuptools.package-data]
+    "lsst.daf.butler" = ["py.typed", "configs/*.yaml", "configs/*/*.yaml"]
+
+    [tool.setuptools.dynamic]
+    version = { attr = "lsst_versions.get_lsst_version" }
+
+This tells ``setuptools`` that the python files are in a ``python/`` directory and what additional non-python files should be included in the distribution.
+
+The license-files section should reflect the specific needs of your package.
+
+When making a `PyPI`_ distribution, the package should work without relying on the EUPS ``$PACKAGE_DIR`` variable being set.
+This means that any supplementary data such as those that would go in a ``config/`` or ``policy/`` directory should instead be included inside the ``python/`` directory and be accessed using the standard package resources APIs.
+These files must then be listed explicitly in the ``package-data`` section of the configuration file.
+
+.. warning::
+
+  Currently ``pex_config`` does not understand how to read a config from a package using package resources.
+  If configs are to be read they can not be read using the usual ``lsst.utils.getPackageDir`` API and must instead use `importlib.resources` APIs directly.
+  We are planning to make this simpler by adding native support into ``pex_config``.
+
+Using GitHub Actions
+====================
+
+If a package is pip-installable it is likely that you will want to build the package in a GitHub action and run the associated tests.
+If your package depends on other science pipelines packages you will want to install those directly from GitHub from the ``main`` branch since there is no guarantee that `PyPI`_ will have the right version.
+The easiest way to do this is to write a ``requirements.txt`` file which has the direct dependencies that should be installed by the build script.
+This file is a simple text file listing packages and versions.
```

Comment:
It may be good to note that this does duplicate information from `pyproject.toml`, and both must be maintained.